### PR TITLE
feat: add eip1271 support

### PR DESCRIPTION
**Pull request by devceline**
- add smart contract wallet detection via RPC call
- accept `rpcUrl` as parameter, this will be used for above detection

### CODE REVIEW

File: packages/core/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "tsconfig/react-library.json",
-  "include": ["."],
+  "include": [".", "../../../../../../tmp/!home!clinenerd!dev!widget-web3inbox!packages!core!src!utils!address.ts~"],
```
[the remark this answers is not in the document]
Comment (devceline) [reply packages/core/tsconfig.json:3]: Ah yep I meant to clean this up before moving out of draft

File: packages/core/src/client/index.ts [hunk 2]
```diff
@@ -37,7 +40,8 @@ export class Web3InboxClient {
   public constructor(
```
Comment (chris13524) [inline packages/core/src/client/index.ts:40]: Should this be a private constructor? Seems like we are ok making a breaking change here

File: packages/core/src/client/index.ts [hunk 3]
```diff
@@ -563,20 +579,30 @@ export class Web3InboxClient {
...
+        .split(":")
+        .slice(-3)
+        .join(":");
+
+      const [chainPrefix, chain, address] = account.split(":");
+
+      const isEip1271Signature = await isSmartWallet(
+        address,
+        `${chainPrefix}:${chain}`,
+        this.notifyClient?.core?.projectId ?? "",
```
Comment (chris13524) [inline packages/core/src/client/index.ts:592]: Defaulting to empty string is not good, it should error

File: packages/core/src/utils/address.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          jsonrpc: "2.0",
+          method: "eth_getCode",
+          params: [address, "latest"],
+          id: Date.now(),
+        }),
+      }
+    ).then((response) => response.json());
+
+    return result;
+  } catch (e) {
```
Comment (chris13524) [inline packages/core/src/utils/address.ts:25]: Can we not surpress `e`? Add it to `Error`

File: packages/core/src/client/index.ts [hunk 5]
```diff
@@ -204,7 +207,8 @@ export class Web3InboxClient {
     projectId: string;
     domain?: string;
     allApps?: boolean;
-    logLevel?: "error" | "info" | "debug"
+    logLevel?: "error" | "info" | "debug";
+    rpcUrl?: string;
```
Comment (chris13524) [inline packages/core/src/client/index.ts:211]: This doesn't seem like right API. This takes any rpcUrl, but it expects it to be a Blockchain API URL and will format it as such.

Instead, this should be a callback that given a chain ID, it returns a URL. By default, that's a callback that generates a URL containing the project ID.

**Review by chris13524 — CHANGES_REQUESTED** [review]

File: packages/core/src/client/index.ts (hunk 2 above)
Comment (devceline) [reply packages/core/src/client/index.ts:40]: This should have been private, and it's not documented to be used. Only the `static` method for `init` is documented

Comment (devceline) [reply packages/core/src/client/index.ts:40]: I'm okay with making breaking changes here, this constructor is undocumented

File: packages/core/src/client/index.ts (hunk 5 above)
Comment (devceline) [reply packages/core/src/client/index.ts:211]: So users of the SDK should supply a callback like:

```
(chainId: string) => `https://their-favorite-rpc/${chainId}`
```

And by default it's
```
(chainId: string) => `https://rpc.walletconnect.com?projectId=${this.projectId}&chainId=${chainId}`
```

That's doable

**Review by chris13524 — APPROVED** [review]

File: packages/core/package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3inbox/core",
-  "version": "1.0.3",
+  "version": "1.1.0",
```
Comment (enesozturk) [inline packages/core/package.json:3]: Do we align this package with notify-client or just coincidence?

Comment (devceline) [reply packages/core/package.json:3]: Just a coincidence, bumped a `minor` version because we added a feature

File: packages/react/package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3inbox/react",
-  "version": "1.0.3",
+  "version": "2.0.0",
```
Comment (enesozturk) [inline packages/react/package.json:3]: Does only React package have major changes? Should it be aligned with Core?

File: packages/react/src/hooks/useNotifications.ts [hunk 8]
```diff
@@ -6,12 +6,12 @@ import { useWeb3InboxClient } from "./useWeb3InboxClient";
 const waitFor = async (condition: () => boolean) => {
   return new Promise<void>((resolve) => {
     setInterval(() => {
-      if(condition()) {
-	resolve()
+      if (condition()) {
```
[the remark this answers is not in the document]
Comment (devceline) [reply packages/react/src/hooks/useNotifications.ts:9]: I just ran `prettier`

[the remark this answers is not in the document]
Comment (chris13524) [reply packages/react/src/hooks/useNotifications.ts:9]: We should setup `prettier` in CI checks

**Pull request merged**